lib/jekyll/site.rb

```diff
@@ -237,7 +237,12 @@ def post_attr_hash(post_attr)
       # array of posts ) then sort each array in reverse order.
       hash = Hash.new { |h, key| h[key] = [] }
       posts.docs.each do |p|
-        p.data[post_attr].each { |t| hash[t] << p } if p.data[post_attr]
+        attr_value = p.data[post_attr]
+        if attr_value.respond_to? :each
+          attr_value.each { |t| hash[t] << p }
+        elsif !attr_value.nil?
+          hash[attr_value] << p
```

Comment:
I think you can use `Kernel#Array` here! It just wraps it in an array in case it's not already an array.

```ruby
Array(p.data[post_attr]).each { |attribute| hash[attribute] << p } if p.data[post_attr]
```